pkg/auth/providers/keycloakoidc/keycloak_provider.go

```diff
@@ -0,0 +1,206 @@
+		}
+	}
+	return princ
+}
+
+func (k *keycloakoidcProvider) GetPrincipal(principalID string, token v3.Token) (v3.Principal, error) {
+	config, err := k.getOIDCConfig()
+	if err != nil {
+		return v3.Principal{}, err
+	}
+	accessToken, err := k.oidcProvider.TokenMGR.GetSecret(token.UserID, token.AuthProvider, []*v3.Token{&token})
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return v3.Principal{}, err
+		}
+		accessToken = token.ProviderInfo["access_token"]
+	}
+	var externalID string
+	parts := strings.SplitN(principalID, ":", 2)
+	if len(parts) != 2 {
+		return v3.Principal{}, errors.Errorf("invalid id %v", principalID)
+	}
+	externalID = strings.TrimPrefix(parts[1], "//")
+	parts = strings.SplitN(parts[0], "_", 2)
+	if len(parts) != 2 {
+		return v3.Principal{}, errors.Errorf("invalid id %v", principalID)
+	}
+	principalType := parts[1]
+	searchType := "users"
+	if principalType == GroupType {
+		searchType = "groups"
+	}
+	acct, err := k.keycloakoidcClient.getFromKeyCloakByID(externalID, searchType, accessToken, config)
+	if err != nil {
+		return v3.Principal{}, err
+	}
+	princ := k.toPrincipal(principalType, acct, &token)
+	return princ, err
+}
+
+func (k *keycloakoidcProvider) TransformToAuthProvider(authConfig map[string]interface{}) (map[string]interface{}, error) {
+	return k.oidcProvider.TransformToAuthProvider(authConfig)
+}
+
+func (k *keycloakoidcProvider) getRedirectURL(config map[string]interface{}) string {
+	return k.oidcProvider.GetRedirectURL(config)
+}
+
+func (k *keycloakoidcProvider) RefetchGroupPrincipals(principalID string, secret string) ([]v3.Principal, error) {
+	return k.oidcProvider.RefetchGroupPrincipals(principalID, secret)
+}
+
+func (k *keycloakoidcProvider) CanAccessWithGroupProviders(userPrincipalID string, groupPrincipals []v3.Principal) (bool, error) {
+	return k.oidcProvider.CanAccessWithGroupProviders(userPrincipalID, groupPrincipals)
+}
+
+func (k *keycloakoidcProvider) userToPrincipal(userInfo *gooidc.UserInfo, info oidc.ClaimInfo) v3.Principal {
+	return k.oidcProvider.UserToPrincipal(userInfo, info)
+}
+
+func (k *keycloakoidcProvider) groupToPrincipal(groupName string) v3.Principal {
+	return k.oidcProvider.GroupToPrincipal(groupName)
+}
+
+func (k *keycloakoidcProvider) toPrincipalFromToken(principalType string, princ v3.Principal, token *v3.Token) v3.Principal {
+	return k.oidcProvider.ToPrincipalFromToken(principalType, princ, token)
+}
+
+func (k *keycloakoidcProvider) saveOIDCConfig(config *v32.OIDCConfig) error {
+	return k.oidcProvider.SaveOIDCConfig(config)
+}
+
+func (k *keycloakoidcProvider) getOIDCConfig() (*v32.OIDCConfig, error) {
+	return k.oidcProvider.GetOIDCConfig()
+}
+
+func (k *keycloakoidcProvider) isThisUserMe(me v3.Principal, other v3.Principal) bool {
+	return k.oidcProvider.IsThisUserMe(me, other)
+}
+
+func (k *keycloakoidcProvider) GetUserExtraAttributes(token *v3.Token) map[string][]string {
+	return k.oidcProvider.GetUserExtraAttributes(token)
+}
```

Comment:
Some of these are needed, to call the OIDC provider, the ones that are not exported should not be here

Reply:
I removed the calls from being exported that did not need to be